pkg/operation/shoot/shoot.go

```diff
@@ -230,6 +230,21 @@ func ComputeTechnicalID(projectName string, shoot *gardencorev1beta1.Shoot) stri
 	return fmt.Sprintf("%s%s--%s", TechnicalIDPrefix, projectName, shoot.Name)
 }
 
+// UnfoldTechnicalID extracts the project and shoot names from Shoot's from the TechnicalID which is used
+// for the name of the namespace
+func UnfoldTechnicalID(technicalID string) (projectName, shootName string) {
+	var technicalIDComponents []string
+	if strings.Contains(technicalID, "--") {
```

Comment:
```suggestion
	if strings.HasPrefix(technicalID, "shoot--") {
```

would be safer I think

Reply:
Also, can you please implement a small unit test here?

Reply:
yes.